Add Geary

mjog opened the pull request:
New Geary 0.12 release no longer depends on WebKit1GTK, so it would be good to get some wider coverage.

TingPing commented on org.gnome.Geary.json:
```diff
@@ -0,0 +1,113 @@
+        "--device=dri",
+
+        /* Bus access */
+        "--socket=session-bus",
```
No reason for broad bus access.

TingPing commented on org.gnome.Geary.json:
```diff
@@ -0,0 +1,113 @@
+        "--talk-name=ca.desrt.dconf", "--env=DCONF_USER_CONFIG_DIR=.config/dconf",
+
+        /* Let view source keep on working as-sis for now. Bug 779311. */
+        "--filesystem=/tmp"
```
Usually the solution to something wanting `/tmp` is to set `TMPDIR=$XDG_CACHE_HOME`: https://github.com/flathub/com.discordapp.Discord/blob/master/com.discordapp.Discord.json#L52

TingPing commented on org.gnome.Geary.json:
```diff
@@ -0,0 +1,113 @@
+    "runtime-version": "3.26",
+    "sdk": "org.gnome.Sdk",
+    "command": "geary",
+    "tags": [],
```
Can just remove empty values.

TingPing commented:
Test build: https://flathub.org/builds/#/buildrequests/2874?redirect_to_build=true

mjog commented:
Thanks for the prompt review @TingPing. I've updated the branch to fix a build issue and address most of your comments.

The one outstanding thing is allowing access to /tmp. I don't believe it is as straight-forward as the Discord case, please have a look at the referenced bug ([b.g.o 779311](https://bugzilla.gnome.org/show_bug.cgi?id=779311) and the [related flatpak issue](https://github.com/flatpak/flatpak/issues/592). We plan on addressing this issue in the next main release, but for now added the exception for access to /tmp to avoid loss of some very useful functionality when running under flatpak.

alexlarsson commented:
New test build here: https://flathub.org/builds/#/builders/1/builds/510

alexlarsson commented:
Created at https://github.com/flathub/org.gnome.Geary

alexlarsson commented:
@mjog I added you to the repo too, can you remove it from the gnome-apps-nightly stable branch ?
